bin/dada2.R

```diff
@@ -1,87 +1,158 @@
+#!/usr/bin/env Rscript
+# -*- coding: utf-8 -*-
+
+# Copyright 2024 EMBL - European Bioinformatics Institute
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+
+# Have to use `box` instead of `library` and `source` so that custom scripts can be loaded when executed by Nextflow
+box::use(tidyverse[...])
+box::use(data.table[...])
+box::use(dada2[...])
+
+# Custom function for tracking reads to their DADA2-generated ASVs (bin/read_asv_tracking.R) 
+box::use(./read_asv_tracking[...])
+# Custom function for automatic truncation of reads based on quality scores (bin/trunc_len_automation.R)
+box::use(./trunc_len_automation[...])
+
+args = commandArgs(trailingOnly=TRUE) # Expects at most three arguments, a prefix, and one fastq for each strand (F and R)
+                                      # If it's a single-end run, then the third argument should not be used
+prefix = args[1] # Prefix
+path_f = args[2] # Forward fastq
+path_r = args[3] # Reverse fastq
+
+# different tax ranks for silva/pr2
+silva_tax_vec = c("Superkingdom", "Kingdom", "Phylum", "Class", "Order", "Family", "Genus", "Species")
+pr2_tax_vec = c("Domain", "Supergroup", "Division", "Subdivision", "Class", "Order", "Family", "Genus", "Species")
+
+# Identify truncLen parameter for filterAndTrim function
+final_where_to_cut_f = trunc_len_automation(path_f)
+if (!is.na(path_r)){
+  final_where_to_cut_r = trunc_len_automation(path_r)
+}
 
-library(tidyverse)
-library(dada2)
-library(data.table)
-
-# Load function for tracking reads to their DADA2-generated ASVs
-source("/hps/software/users/rdf/metagenomics/service-team/users/chrisata/asv_gen/bin/read_asv_tracking.R")
-
-args = commandArgs(trailingOnly=TRUE) # Expects thre arguments, one fastq for each strand (F and R), and a prefix
-
-path_f = args[1] # Forward fastq
-path_r = args[2] # Reverse fastq
-prefix = args[3] # Prefix
+# Do some quality filtering
```

Comment:
This is ASV-specific QC that needs to be done annoyingly since all the reads need to be the same length, I think we're happy with inputs being slightly different for ASV calling vs the standard analysis we do though so I think it's okay